### ISSUE TITLE
Dialog on delete?

### ISSUE THREAD

**Issue by unclebay143** [BALKANGraph/FamilyTreeJS#150]
Similar to https://code.balkan.app/org-chart-js/dialog-on-save#JS

**Comment by ZornitsaPesheva** [comment]
For FamilyTree JS we use only onUpdateNode to update the database:
https://balkan.app/FamilyTreeJS/Docs/GettingStarted#database

**Comment by unclebay143** [comment]
> For FamilyTree JS we use only onUpdateNode to update the database: https://balkan.app/FamilyTreeJS/Docs/GettingStarted#database

Does that mean we cannot display a confirmation modal for delete? And can't prevent accidental delete through confirmation?

Deleted members are also deleted from the tree before the `onUpdateNode` is triggered

**Comment by ZornitsaPesheva** [comment]
No, you can have the same dialog, but you need to use on the onUpdateNode.
You have the deleted nodes in args.

**Comment by unclebay143** [comment]
> No, you can have the same dialog, but you need to use on the onUpdateNode. You have the deleted nodes in args.

that's the point, we can't intercept it, even if we show a confirmation modal, it gets deleted before the modal shows
for instance my code looks like this.
![image](https://github.com/user-attachments/assets/acf3da04-df23-45bd-93f5-4b41f9a86c2e)

**Comment by unclebay143** [comment]
Thanks @ZornitsaPesheva, able to achieve my thought now, the overall code now is as below:

```
familyTree.current.onUpdateNode(
  (updatedTreeData: { removeNodeId: string }) => {
    const isDeleteAction = !!updatedTreeData.removeNodeId;
    if (isDeleteAction && !confirm("Do you want to delete?")) {
      return false;
    }
    saveTree(updatedTreeData);
  }
);
```
